### PR TITLE
Anaerobic Lagoon VS update

### LINKED ISSUE #1368: Fix update of VS in anaerobic digestion and lagoon treatment
This issue is related to Issue #1270: Volatile solids updates in the manure module. Issue 1270 refers primarily to slurry storage outdoor and underfloor methods, whereas this issue will cover anaerobic digestion treatment file, as there are some differences in how volatile solids (VS) should be updated.

## Expected Behavior
In the anaerobic digestion portion of the ADL treatment, we expect that a portion of the manure VS loaded into the digester will be destroyed through the anaerobic fermentation process that occurs inside the digester. Therefore, the VS content of the manure (effluent) leaving the digester and subsequently entering the anaerobic lagoon should be lower than the manure from the reception pit. This ultimately results in lower methane emissions from the anaerobic lagoon.

Additionally, if a separator is employed between the digester and lagoon (i.e., anaerobic digestion and lagoon with separator treatment), the VS content of manure entering the separator (and other nutrients, though this issue focuses on VS) should be updated to reflect VS loss during anaerobic digestion as well. The diagram below depicts both situations.

For example, say a pen's manure VS on one day is 100 kg (already accounting for any emissions lost in housing). Assume we lose 35% of the total VS through anaerobic digestion - the VS content of the manure leaving the digester is now 65 kg. Next, we optionally use a rotary screen separator and remove 40% of the VS - the VS content of the manure leaving the separator is now 39 kg. Thus, the manure entering the anaerobic lagoon on this day contains 39 kg of VS. If we had not used the separator, the manure VS would be 65 kg. 

## Current Behavior
Currently, there is no loss of VS or TS accounted for in the anaerobic digestion treatment file. Additionally, the individual VS fractions are not tracked and updated separately.

## Possible Solution
Please see [the first page of this doc ](https://docs.google.com/document/d/1xKSRiwmKHE0tPmK0C0dkyzXynBt-DBimWSkwsSt4KAU/edit)for further detail on how VS loss in anaerobic digestion should be calculated and updated. The subsequent pages contain some information and subject matter discussion. 

The PR to address this issue should:
1. Institute tracking of degradable and non-degradable volatile solids (VSd and VSnd) in the anaerobic digestion treatment file, and removes tracking of total volatile solids.
2. Institute a conversion factor to convert from CH4 generated to CH4+CO2 generated, i.e., total VSd loss/reduction during anaerobic digestion.
3. Update the content of VSd and total solids in manure effluent leaving the digester, which becomes the input to either a separator followed by anaerobic lagoon, or become the input to the lagoon directly. 

## Diagram
image

### PR DESCRIPTION
**Pull request by allisterakun**
This PR addresses the changes requested in this design doc: [Anaerobic digestion VS updates](https://docs.google.com/document/d/1xKSRiwmKHE0tPmK0C0dkyzXynBt-DBimWSkwsSt4KAU/edit)
### Context

Issue(s) closed by this pull request: closes #1368.

- [ ] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.

### What

1. Update the input parameter of `GasEmissionCalculator.methane_volume_via_Chen_equation()` to take in VSd instead of total VS.
2. Change the VSd input for `GasEmissionCalculator.methane_volume_via_Chen_equation()` in `AnaerobicDigestion` to use the daily input value (`self._current_manure_treatment_daily_input.liquid_manure_total_degradable_volatile_solids`) instead of manure handler output value (`self._manure_handler_daily_output.liquid_manure_total_degradable_volatile_solids`). @elle-adnreen suggested this due the the possibility of having separation before digestion. **Please confirm this is correct!**
3. Track VSd and total solids in the daily output of `AnaerobicDigestion`. (Input value - (biogas * AD_METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO))
4. Pass through VSnd from input and report total_VS=VSd+VSnd.
5. Rename `GasEmissionConstants.METHANE_DENSITY` to `GasEmissionConstants.AD_METHANE_DENSITY` and update the value to 0.629. 
6. Introduce a new constant `GasEmissionConstants.AD_METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO = 2.86`.
7. Move `ManureConstants.METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO` into `GasEmissionConstants` and update all occurrences.

### Test plan

All affected tests have been updated

### CODE REVIEW

**Review by tomhuhh — APPROVED** [review]
Thanks Allister! Great work! I believe this is in alignment with my expectations. 

For my testing, I used scenario 6 where there is a separator before the anaerobic digester but not after. I specifically checked the degradable and non degradable volatile solids at the reception pit, at the separator, and at the digester. At each step I noticed a decrease in the amount of degradable volatile solids but no change from separator to digester for non degradable volatile solids, which means to me that the function is working. Please find the visualization below. 

Degradable volatile solids: 
default_reception-pit-total-degradable-solids-04-Apr-2024_Thu_09-48-20

default_manure-separator-total-degradable-solids-04-Apr-2024_Thu_09-48-20

default_ad-degradable-solids-04-Apr-2024_Thu_09-48-20

Non degradable volatile solids:  
default_reception-pit-total-non-degradable-solids-04-Apr-2024_Thu_09-48-20

default_manure-separator-total-non-degradable-solids-04-Apr-2024_Thu_09-48-20

default_ad-non-degradable-solids-04-Apr-2024_Thu_09-48-20

Thanks!

**Comment by elle-andreen** [comment]
This PR also looks good to me. I did let Allister know that anaerobic lagoon alone seemed to be double-accumulating storage methane values, but that should be an easy fix.

Mathematically, TS/VS/VSd/VSnd adjustments seem to be occurring where they should be. I also did a similar comparison to Haowen but with slightly different metrics. I ran one simulation with 4 lactating cow pens with each treatment combo of interest. The average kg CH4/animal/year from storage decreases as expected with the use of anaerobic digestion and separator(s):

- Lagoon alone (scenario 0): **42.17 kg CH4/cow/year** (corrected for double-accumulating)
- Anaerobic digestion and lagoon with separator _before_ AD (scenario 6): **38.7 kg CH4/cow/year**
- Anaerobic digestion and lagoon, with separator _after_ AD (scenario 7): **34.4 kg CH4/cow/year.** 
- Anaerobic digestion and lagoon, with separator _before and after_ AD (scenario 10): **20.2 kg CH4/cow/year**. The CH4 reduction here may be a little inflated - separation efficiency can decrease when the TS content of manure is low, but we assume the same separation efficiencies pre and post digestion, even though digestate will definitely be lower TS than manure entering digestion. 

Not sure if we want to just fix the anaerobic lagoon double-accumulation in this PR or not (if that is indeed happening and it's not just me!). If we do, I'll wait to approve, and if not, I think this is ready to merge!

File: RUFAS/routines/manure/constants_and_units/gas_emission_constants.py [hunk 1]
```diff
@@ -56,8 +56,22 @@ class GasEmissionConstants:
...
+
+    METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO: float = 9.25
+    """
+    The mass conversion factor from methane to methane and carbon dioxide based on a molar ratio of 1:3
+    (methane : carbon dioxide).
+    """
+
+    AD_METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO: float = 2.86
+    """
+    The mass conversion factor from methane to methane and carbon dioxide in anaerobic digestion (kg CH4 / kg CO2)
```
Comment (ew3361zh) [inline RUFAS/routines/manure/constants_and_units/gas_emission_constants.py]:
```suggestion
    The mass conversion factor from methane to methane and carbon dioxide in anaerobic digestion (kg CH4 / kg CO2).
```

**Review by ew3361zh — APPROVED** [review]
Nice work, Allister! If the science is checking out, it should be good to go.

**Pull request merged**